compiler/rustc_parse/src/errors.rs

```diff
@@ -2378,6 +2378,16 @@ pub(crate) struct ExpectedCommaAfterPatternField {
     pub span: Span,
 }
 
+#[derive(Diagnostic)]
+#[diag(parse_unexpected_expr_in_pat)]
+pub(crate) struct UnexpectedExpressionInPattern {
+    #[primary_span]
+    #[label]
+    pub span: Span,
+    pub is_bound: bool,
+    pub is_methodcall: bool,
```

Comment:
Minor: Could you replace all occurrences of *methodcall* (no space) with *method call* in prose and `method_call` in code? In English, noun-noun compounds are usually spaced esp. for jargon.